### PR TITLE
ci: Add bento check for pull requests

### PR DESCRIPTION
**Pull request by underyx**
Heya 👋

I just made https://github.com/returntocorp/bento-action which will mark PRs as failed if they introduce any new issues. We wanted to dogfood it and test it on all r2c projects, so I'd appreciate a merge 🙃

### CODE REVIEW

**Comment by underyx** [comment]
`r2c.flask` is not returning output, any idea why? Otherwise I'll investigate tomorrow.

**Comment by nbrahms** [comment]
> `r2c.flask` is not returning output, any idea why? Otherwise I'll investigate tomorrow.

@ulziibay you ran into this issue a while back, right? How did you fix it?

I seem to remember something about dueling `flake8` installations...

**Comment by ulziibay** [comment]
@underyx  thanks for the PR. @nbrahms It is hard for me to debug the issue without seeing the last.log file on the https://github.com/returntocorp/sgrep/pull/149/checks?check_run_id=439673935. I wonder if its versioning issue. Bento: 0.8.2 should be the latest bento-cli we should be using. Better yet if the workflow optionally accepted which bento version to use in the CI, that would be great.

A few notes on the details:
1) Can we make sure we are installing latest bento-cli package by pinning the version of bento-cli in https://github.com/returntocorp/bento-action/blob/master/Pipfile
2) Can we make this workflow emit the bento -version as a step before 1 and 2?
3) Can we  unhide the stderr, e.g. `bento init` as opposed to `bento init &> /dev/null`

**Comment by underyx** [comment]
Thanks for the feedback @ulziibay! Let me address all these:

> Can we make sure we are installing latest bento-cli package by pinning the version of bento-cli in https://github.com/returntocorp/bento-action/blob/master/Pipfile

Could you explain further? I thought pinning with pipenv was meant to occur in the lockfile, here: https://github.com/returntocorp/bento-action/blob/50c2790130cd7b98160220fe3047432f611c17cd/Pipfile.lock#L40-L47

But it's the first time I'm using pipenv (poetry guy myself 🙃) so do let me know if/where I'm wrong!

> Can we make this workflow emit the bento -version as a step before 1 and 2?

Good idea, will do!

> Can we unhide the stderr, e.g. `bento init` as opposed to `bento init &> /dev/null`

Are you sure about this? I intentionally hid it as `bento init` outputs *a lot* of text and makes it difficult for end users to navigate the output if they just want to see the results. Perhaps if I set it to output stderr in debug mode only? (see https://help.github.com/en/actions/configuring-and-managing-workflows/managing-a-workflow-run#enabling-debug-logging)

> It is hard for me to debug the issue without seeing the last.log file

And I could print out this file as well if you'd like in the same debug mode. Sounds good?

**Comment by underyx** [comment]
@ulziibay missed this one

> Better yet if the workflow optionally accepted which bento version to use in the CI, that would be great.

I think @dlukeomalley specifically wanted to force latest for now and also to not add any extra features until somebody specifically needs it. Reason being, the action is a tool to help people stay on the latest version of Bento. An option to pin might encourage people to do just that, so I'd leave it out for now if that's okay.

**Comment by ulziibay** [comment]
@nbrahms @underyx so what is happening is bento is trying to run the flake8 plugin on files in tests/TODO which contain malformed .py files e.g. tests/TODO/bad.py 
```python
def foo
  return 1
```
I recommended we track this on the bento python tool realm as an issue!

**Comment by dlukeomalley** [comment]
Chatting with Bence a bit, I think it's expected / correct behavior for us to fail in this case. We may need a better error message though that suggests adding this file or directory to `.bentoignore`. If this were a production file, this could have become a runtime issue.

To fix the problem for sgrep in particular, let's add `tests/TODO/bad.py` to `.bentoignore`.

Does anyone have suggestions about how better wrap or display this error on Bento? I can open a ticket...

**Comment by underyx** [comment]
I think it should be okay now after blanket ignoring the tests directory! I tried only ignoring `bad.py` but the issue persisted, and I couldn't figure out from `last.log` which other file is causing it. If someone knows how to limit the ignore to necessary test files only, let me know.

The action is still failing because we run bento on the main branch (`develop`) first, where the ignores aren't yet configured. It should work with all other PRs after being merged.

**Comment by dlukeomalley** [comment]
> The action is still failing because we run bento on the main branch (develop) first, where the ignores aren't yet configured. It should work with all other PRs after being merged.

That's an interesting edge case. Does the Bento action use the same configs for both analyses?

**Comment by brendongo** [comment]
Why did bento configuration files need to be added for the github action to work?

**Comment by underyx** [comment]
> Why did bento configuration files need to be added for the github action to work?

@brendongo it's due to https://github.com/returntocorp/bento-core/issues/718. 

> Does the Bento action use the same configs for both analyses?

It uses the config from `develop` (i.e. default ignores only) for the first analysis. Hence the error before that will stay until this PR is merged.

**Comment by underyx** [comment]
I ended up digging further and tracking down the exact cause of the failure: https://github.com/returntocorp/bento-core/issues/718#issuecomment-591470536

Thought it made sense to change the crash-inducing test in this repo, so I've done that and reverted the bento config changes.

File: .github/workflows/bento.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: Bento
```
Comment (dlukeomalley) [inline .github/workflows/bento.yml]: Is this still up to date with the latest Bento Check configs? https://github.com/marketplace/actions/bento-check

Comment (underyx) [reply .github/workflows/bento.yml]: Right, no, I'll go update the name!

**Comment by DrewDennison** [comment]
closing for now

**Comment by underyx** [comment]
@DrewDennison could you provide the reason? Is this related to the sgrep-rules repo?

**Comment by DrewDennison** [comment]
@underyx no just that there were merge conflict and it's probably easier to just make a clean pr since we moved a bunch of stuff around this week. Please open a new PR and I'll get it merged it

**Comment by underyx** [comment]
Ah I see! Could you reopen the PR? I rebased my branch again and I think it should be ok to merge now.

**Comment by nbrahms** [comment]
I think because the develop base branch was rebased, you'll need to make a new PR :(

**Pull request closed without merging**